Tests/WebSocketKitTests/WebSocketKitTests.swift

```diff
@@ -291,6 +328,122 @@ final class WebSocketKitTests: XCTestCase {
         try server.close(mode: .all).wait()
     }
 
+    func testProxy() throws {
+        let promise = elg.next().makePromise(of: String.self)
+
+        let localWebsocketBin: WebsocketBin
+        let verifyProxyHead = { (ctx: ChannelHandlerContext, requestHead: HTTPRequestHead) in
+            XCTAssertEqual(requestHead.uri, "ws://apple.com/:\(ctx.localAddress!.port!)")
```

Comment:
Just to confirm this is a simulated proxy and not something we're going out on the network for?

Reply:
Yeah, we do a somewhat convoluted dance to insert and remove channel handlers to mimic a proxy e.g. intercepting the `CONNECT` request.